src/librustc_resolve/lib.rs

```diff
@@ -3127,6 +3127,22 @@ impl<'a, 'crateloader: 'a> Resolver<'a, 'crateloader> {
                         }
                         return (err, candidates);
                     },
+                    (Def::Struct(_), PathSource::Expr(Some(parent))) if ns == ValueNS => {
+                        match parent.node {
+                            ExprKind::MethodCall(ref path_assignment, _)  => {
+                                err.span_suggestion_with_applicability(
+                                    parent.span,
+                                    "incorrect associated function syntax",
+                                    format!("try {}::{}()?",
```

Comment:
@dsciarra The [third argument to the `span_suggestion_` methods](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_errors/struct.Diagnostic.html#method.span_suggestion_with_applicability) should be a code snippet to replace the span passed as the first argument. We're going to need something more like 

```rust
err.span_suggestion_with_applicability(
    parent.span,
    "use `::` to access an associated function",
    format!("{}::{}", path_str, path_assignment.ident)
);
```

(But this probably isn't literally correct as written. In the UI test output, it looks like our span includes the call args (here, just `()`), which we don't want to replace.)

Reply:
If you need to get part of the literal code, you can use `Codemap::span_to_snippet` to get it, like done here:

https://github.com/rust-lang/rust/blob/f1694eac74216da22b395a5589b8d080ccd5a2ec/src/librustc_typeck/check/demand.rs#L270